### PR TITLE
Batch comparison: Add extra batches once per round

### PR DESCRIPTION
**Pull request by nikhilb4a**
https://github.com/votingworks/arlo/issues/2324

Georgia noticed in their recent audit that more extra batches were selected than needed, and this was because extra batches where being added once per contest instead of once per round. Batches audited for one contest are already audited for the other contest, so we don't need extra batches added for each.

This PR makes a low-lift effort to refactor the extra batch selection to make it once per round instead of per contest.

This [subsequent PR](https://github.com/votingworks/arlo/pull/2344) suggests data model changes to move extra batches out of the experimental state and into official support, but given we are close to the audit day (Thursday), I think it's safer to just go with the lower lift change for now. 

Testing: 

Will do a manual test with the GA extra batch flag turned on for multi-contest.
Will add a multi-contest extra batch test.

### CODE REVIEW

**Comment by nikhilb4a** [comment]
@jonahkagan I put together this simpler fix for this ticket, as my other PR (mentioned in PR description) ended up being having more surface area than I expected, and I don't feel as confident about getting it done by Thursday. Can you take a look if this approach seems reasonable for now, and then I'll finalize the logic/testing?

**Review by jonahkagan — APPROVED** [review]
Approach seems reasonable to me. The comment from copilot seems worth looking into

File: server/api/shared.py [hunk 1]
```diff
@@ -681,12 +680,48 @@ def compute_sample_batches_for_contest(
...
+
+    extra_batches: list[BatchDraw] = []
+
     if is_enabled_sample_extra_batches_by_counting_group(election) and round_num == 1:
         rand = random.Random(str(election.random_seed))
-        for jurisdiction in contest.jurisdictions:
+        for jurisdiction in jurisdictions:
+            representative_contest_id = jurisdiction_id_to_contest_id[jurisdiction.id]
             batch_key_to_num_ballots = {
                 (jurisdiction.name, batch.name): batch.num_ballots
```
[the remark this answers is not in the document]
Comment (nikhilb4a) [reply server/api/shared.py:735]: @jonahkagan I think the comment is correct, though it seems to predate this PR. sampled_batch_results() gets all batches which have batch tally sheets for this contest, and then when excluding non-rla batches, it only filters out extra batches that were tagged for this contest. But if contest B pulls in an extra batch, there will be a tally sheet with contest A for that batch, which ends up included in the sampled_rows. 

I think this [fixes](https://github.com/votingworks/arlo/pull/2346/commits/e2d6e1bf098671c9b57cf24f1d08fbf833b4164b) it, by removing all extra batches except any that also live in the rla set (if a batch gets chosen as extra in first round, but part of sample in second round)

File: server/api/shared.py [hunk 2]
```diff
@@ -227,7 +235,7 @@ def sampled_batch_results(
         results = {
             batch_key: result
             for batch_key, result in results.items()
-            if batch_key not in extra_batch_keys
+            if batch_key not in (extra_batch_keys - rla_sampled_batch_keys)
```
Comment (jonahkagan) [inline server/api/shared.py:238]: Why do we need to subtract out `rla_sampled_batch_keys` here? I didn't think it was possible for an RLA-sampled batch to also be used as an extra batch

Comment (nikhilb4a) [reply server/api/shared.py:238]: That was considering a multi-round audit that has a batch selected as an extra in round 1, but then part of the sample in round 2, as I think that sequencing can happen

Comment (jonahkagan) [reply server/api/shared.py:238]: Oh, interesting edge case. Cool

**Comment by nikhilb4a** [comment]
Did a manual test with the GA election from May and saw intended results. In my local snapshot, I cleared the batch tallies and undid the round start. Upon redrawing, every jurisdiction saw their number of extra batches stay the same or go down. Looked into some jurisdictions with the largest changes (along with some Claude investigation) and they were correctly sampled.

Feeling confident enough to merge.
